lib/child_process.js

```diff
@@ -473,7 +473,7 @@ function spawnSync(/*file, args, options*/) {
 
   // Validate maxBuffer, if present.
   if (options.maxBuffer != null &&
-      !(Number.isInteger(options.maxBuffer) && options.maxBuffer >= 0)) {
+      !(typeof options.maxBuffer === 'number' && options.maxBuffer >= 0)) {
     throw new TypeError('"maxBuffer" must be an unsigned integer');
```

Comment:
Error message needs to be updated to reflect the new constraint (It now accepts any positive number, not just unsigned integer).